app/controllers/users_controller.rb

```diff
@@ -1,7 +1,57 @@
+# frozen_string_literal: true
+
 class UsersController < Clearance::UsersController
-  private
+  before_action :current_user, only: [:show, :edit, :update]
+  before_action :check_for_cancel, only: [:edit, :update]
+
+  # Could be useful for a future issue
+  # before_action :another_user
+
+  # def another_user
+  #   unless user_url.end_with?(current_user.id.to_s)
+  #     redirect_to request.referer
+  #   end
+  # end
+
+  def check_for_cancel
+    if params[:commit] == "Cancel"
+      redirect_to user_url
+    end
+  end
+
+  def index
+    @users = User.all
```

Comment:
Chatted offline: We are going to nuke this. It's not needed at all

Reply:
Ready 👍